src/geometry/manifold/mfderiv.lean

```diff
@@ -1227,6 +1227,83 @@ lemma mfderiv_within_const (hxs : unique_mdiff_within_at I s x) :
 
 end const
 
+section arithmetic
+/-! #### Arithmetic
+
+Note that in the in `has_mfderiv_at` lemmas there is an abuse of the defeq between `E'` and
+`tangent_space 𝓘(𝕜, E') (f z)` (similarly for `g',F',p',q'`). In general this defeq is not
+canonical, but in this case (the tangent space of a vector space) it is canonical.
+ -/
+
+variables { z : M} {F' : Type*} [normed_field F'] [normed_algebra 𝕜 F']
+{f g : M → E'} {p q : M → F'}
+{f' g' : tangent_space I z →L[𝕜] E'}
+{p' q' : tangent_space I z →L[𝕜] F'}
+
+lemma has_mfderiv_at.add (hf : has_mfderiv_at I 𝓘(𝕜, E') f z f')
+  (hg : has_mfderiv_at I 𝓘(𝕜, E') g z g') : has_mfderiv_at I 𝓘(𝕜, E') (f + g) z (f' + g') :=
+⟨hf.1.add hg.1, hf.2.add hg.2⟩
+
+lemma mdifferentiable_at.add (hf : mdifferentiable_at I 𝓘(𝕜, E') f z)
+  (hg : mdifferentiable_at I 𝓘(𝕜, E') g z) : mdifferentiable_at I 𝓘(𝕜, E') (f + g) z :=
+(hf.has_mfderiv_at.add I hg.has_mfderiv_at).mdifferentiable_at
+
+lemma mdifferentiable.add (hf : mdifferentiable I 𝓘(𝕜, E') f) (hg : mdifferentiable I 𝓘(𝕜, E') g) :
+  mdifferentiable I 𝓘(𝕜, E') (f + g) :=
+λ x, (hf x).add I (hg x)
+
+lemma has_mfderiv_at.mul (hp : has_mfderiv_at I 𝓘(𝕜, F') p z p')
+  (hq : has_mfderiv_at I 𝓘(𝕜, F') q z q') : has_mfderiv_at I 𝓘(𝕜, F') (p * q) z
+  (((((written_in_ext_chart_at I 𝓘(𝕜, F') z) p ((ext_chart_at I z) z)) • q') : E →L[𝕜] F') +
+  (written_in_ext_chart_at I 𝓘(𝕜, F') z q ((ext_chart_at I z) z) • p' :  E →L[𝕜] F' )) :=
```

Comment:
I know you took this from `has_fderiv_at.mul`, but I don't see why `•` is used -- can it just be `*`?  Also, can the long term 
```
written_in_ext_chart_at I 𝓘(𝕜, F') z p ((ext_chart_at I z) z)
```
just be `p z`?

Reply:
Hmm it doesnt seem to like either of those changes. `*` just complains about wanting `has_mul (E →L[𝕜] F')` and `p z` just gives me a deterministic timeout.

Reply:
This works for me:
```lean
lemma has_mfderiv_at.mul (hp : has_mfderiv_at I 𝓘(𝕜, F') p z p')
  (hq : has_mfderiv_at I 𝓘(𝕜, F') q z q') :
  has_mfderiv_at I 𝓘(𝕜, F') (p * q) z (p z • q' + q z • p' : E →L[𝕜] F') :=
⟨hp.1.mul hq.1, by simpa only with mfld_simps using hp.2.mul hq.2⟩
```
Indeed the `•` is necessary, though -- I had misunderstood.  We are taking the scalar product of an element of `F'` with an element of `E →L[𝕜] F'`.

Reply:
Ah nice I should've tried the `simpa`. I've updated it now :)